dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/RouterChain.java

```diff
@@ -63,19 +64,16 @@
      */
     private volatile List<Router> builtinRouters = Collections.emptyList();
 
-    private volatile List<StateRouter<T>> builtinStateRouters = Collections.emptyList();
-    private volatile List<StateRouter<T>> stateRouters = Collections.emptyList();
+    private volatile StateRouter<T> headStateRouter;
+
+    private volatile List<StateRouter<T>> stateRouters;
```

Comment:
I cannot tell the design purpose of keeping both `RouterChain.stateRouters` and `StateRouter.setNextRouter()`.

We already have a router chain with the following look:
`HeaderRouter` -> `XxxRouter1` -> `XxxRouter2` -> `TailRouter`
All functions like 'notify', 'route', 'destroy', 'getSnapshot' can be started from HeaderRouter and run automatically to the end following the chain.

So what's the necessity of the list attribute in RouterChain hosting [XxxRouter1, XxxRouter2]

Reply:
`stateRouters` is for unit test and qos management perpose.